subcommands/keys/tuf_updates_rotate_offline_key.go

```diff
@@ -185,16 +185,20 @@ func doTufUpdatesRotateOfflineTargetsKey(cmd *cobra.Command) {
 			subcommands.SliceRemove(curCiRoot.Signed.Roles["targets"].KeyIDs, onlineTargetsId))
 		subcommands.DieNotNil(err)
 	}
-	newTargetsSigs, err := signProdTargets(factory, newKey,
-		func(tag string, targets client.AtsTufTargets) bool {
-			for _, sig := range targets.Signatures {
-				if sig.KeyID == oldestKey.Id {
-					return false
-				}
+
+	targetsMap, err := api.ProdTargetsList(factory, false)
+	subcommands.DieNotNil(err, "Failed to fetch production targets:")
+outerLoop:
+	for tag, targets := range targetsMap {
+		for _, sig := range targets.Signatures {
```

Comment:
Just IMO, having the internal loop in some function (local scope) would make it a bit cleaner.

Reply:
I extracted this thing into its own function in one of the last commits: https://github.com/foundriesio/fioctl/pull/352/commits/d1d10684cbf392b9f2bf2758ca6117e63b0218ac.

It is not exactly what you asked for; but it is rather simple now.

Reply:
well, I don't like that idiom. Just don't like blindly following "rules"/idioms" :).
But, it's ok in this case, just imho.

Reply:
I wanted to break an _established rule to never use goto_... so you tell me this. OMG LOL.